apps/builder/app/builder/features/style-panel/sections/transitions/transitions.tsx

```diff
@@ -14,44 +13,66 @@ import { CollapsibleSectionRoot } from "~/builder/shared/collapsible-section";
 import type { SectionProps } from "../shared/section";
 import { getDots } from "../../shared/collapsible-section";
 import { PropertyName } from "../../shared/property-name";
-import { LayersList } from "../../style-layers-list";
 import { $selectedOrLastStyleSourceSelector } from "~/shared/nano-states";
 import { TransitionContent } from "./transition-content";
 import {
-  getTransitionProperties,
   deleteTransitionProperties,
   addDefaultTransitionLayer,
-  deleteTransitionLayer,
-  editTransitionLayer,
-  swapTransitionLayers,
-  hideTransitionLayer,
-  convertIndividualTransitionToLayers,
+  findTimingFunctionFromValue,
 } from "./transition-utils";
-import type { StyleProperty } from "@webstudio-is/css-engine";
-import { getStyleSource } from "../../shared/style-info";
+import {
+  toValue,
+  type StyleValue,
+  type StyleProperty,
+} from "@webstudio-is/css-engine";
+import { humanizeString } from "~/shared/string-utils";
+import { RepeatedStyle } from "../../shared/repeated-style";
+import { getStyleSource, type StyleInfo } from "../../shared/style-info";
 
 const label = "Transitions";
-const initialTransition = "opacity 200ms ease 0s";
 export const properties = (
   transitionLongHandProperties as unknown as Array<StyleProperty>
 ).filter((property) => property !== "transitionBehavior");
 
+const getLayer = (value: undefined | StyleValue, index: number) =>
+  value?.type === "layers" ? value.value[index] : undefined;
+
+const getLayerLabel = ({
+  style,
+  index,
+}: {
+  style: StyleInfo;
+  index: number;
+}) => {
+  // show label without hidden replacement
+  const propertyLayer = getLayer(style.transitionProperty?.value, index);
+  const property = humanizeString(
+    toValue(propertyLayer ? { ...propertyLayer, hidden: false } : undefined)
+  );
+  const duration = toValue(getLayer(style.transitionDuration?.value, index));
```

Comment:
Since, it's going to be `getLayer` approach using index now. And i see the transition extractor is not being used anymore I think we can delete the extractor too in this PR with it's tests.

And use the same approach here too
https://github.com/webstudio-is/webstudio/blob/main/apps/builder/app/builder/features/style-panel/sections/transitions/transition-utils.ts#L291

Reply:
Still used in transition popover. Would require more refactoring. Here focused on the list specifically.